### PR TITLE
Feature/gh 1339 wc completion screen

### LINKED ISSUE #1339: Create the WalletConnect completion HUD
Also, show WalletConnect completion HUD after connection and request handling

[wc-logo.pdf](https://github.com/gnosis/safe-ios/files/3978601/wc-logo.pdf)

### PR DESCRIPTION
**Pull request by DmitryBespalov**
closes #1339 

- Showing the completion screen on the success of connection, and the success/reject of transaction

### CODE REVIEW

File: MultisigWallet/MultisigWalletDomainModel/WalletConnect/WCSession.swift [hunk 1]
```diff
@@ -37,14 +37,16 @@ public struct WCClientMeta: Codable {
 
 }
 
-public struct WCDAppInfo: Codable {
```
Comment (sche) [inline MultisigWallet/MultisigWalletDomainModel/WalletConnect/WCSession.swift]: why did you remove Codable here?

File: MultisigWallet/MultisigWalletImplementations/Repositories/DBWalletConnectSessionRepository.swift [hunk 2]
```diff
@@ -59,17 +59,43 @@ extension MultisigWalletDomainModel.WCURL {
 
 }
 
-extension WCDAppInfo {
+extension WCDAppInfo: Codable {
```
Comment (sche) [inline MultisigWallet/MultisigWalletImplementations/Repositories/DBWalletConnectSessionRepository.swift]: I see.

File: MultisigWallet/MultisigWalletDomainModel/WalletConnect/WCSession.swift (hunk 1 above)
Comment (DmitryBespalov) [reply MultisigWallet/MultisigWalletDomainModel/WalletConnect/WCSession.swift]: because it was producing compile-generated implementations and was not picking up implementations in the protocol extension in another place

File: SafeUIKit/SafeAppUI/MainFlow/IncomingTransactionFlowCoordinator.swift [hunk 3]
```diff
@@ -41,11 +41,18 @@ class IncomingTransactionFlowCoordinator: FlowCoordinator {
                 guard let `self` = self else { return }
                 self.onBackButton?()
                 self.pop()
+                self.showWCCompletion()
```
Comment (sche) [inline SafeUIKit/SafeAppUI/MainFlow/IncomingTransactionFlowCoordinator.swift]: why do we need to show WC_completion onBack?

Comment (DmitryBespalov) [reply SafeUIKit/SafeAppUI/MainFlow/IncomingTransactionFlowCoordinator.swift]: this is the "Reject" button

File: safe/safe/AppDelegate.swift [hunk 4]
```diff
@@ -145,6 +145,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate, Resettable {
             return false
         }
         components.scheme = "wc"
+        components.queryItems?.append(URLQueryItem(name: "isMobile", value: "true"))
```
Comment (sche) [inline safe/safe/AppDelegate.swift]: I have one general question: do we need this at all? Why can't we always act as if it is mobile?

**Review by sche — APPROVED** [review]

File: safe/safe/AppDelegate.swift (hunk 4 above)
Comment (DmitryBespalov) [reply safe/safe/AppDelegate.swift]: it is a valid question, I'll check with Tobi

**Review by sche — APPROVED** [review]
